include/vcpkg/build.h

```diff
@@ -319,15 +319,15 @@ namespace vcpkg::Build
         BuildPolicies policies;
     };
 
-    BuildInfo read_build_info(const Filesystem& fs, const path& filepath);
+    BuildInfo read_build_info(const Filesystem& fs, const Path& filepath);
 
     struct AbiEntry
     {
         std::string key;
         std::string value;
 
         AbiEntry() = default;
-        AbiEntry(const std::string& key, const std::string& value) : key(key), value(value) { }
+        AbiEntry(StringView key, StringView value) : key(key.data(), key.size()), value(value.data(), value.size()) { }
```

Comment:
```suggestion
        AbiEntry(StringView key, StringView value) : key(key.to_string()), value(value.to_string()) { }
```

Reply:
That would result in an unnecessary extra string move construction.

Reply:
Hmmmm.... I guess NRVO saves it? https://gcc.godbolt.org/z/7sWjf7ezb

```
this$ = 48
sv$ = 56
__formal$ = 64
X::X(string_view,use_to_string_tag) PROC    ; X::X, COMDAT
$LN22:
        push    rbx
        sub     rsp, 32                             ; 00000020H
        mov     r8, QWORD PTR [rdx+8]
        xor     eax, eax
        mov     rdx, QWORD PTR [rdx]
        mov     rbx, rcx
        mov     QWORD PTR [rcx], rax
        mov     QWORD PTR [rcx+16], rax
        mov     QWORD PTR [rcx+24], 15
        mov     BYTE PTR [rcx], al
        call    std::basic_string<char,std::char_traits<char>,std::allocator<char> > & std::basic_string<char,std::char_traits<char>,std::allocator<char> >::assign(char const * const,unsigned __int64) ; std::basic_string<char,std::char_traits<char>,std::allocator<char> >::assign
        mov     rax, rbx
        add     rsp, 32                             ; 00000020H
        pop     rbx
        ret     0
X::X(string_view,use_to_string_tag) ENDP    ; X::X

this$ = 48
sv$ = 56
__formal$ = 64
X::X(string_view,use_data_size_tag) PROC    ; X::X, COMDAT
$LN18:
        push    rbx
        sub     rsp, 32                             ; 00000020H
        mov     r8, QWORD PTR [rdx+8]
        xor     eax, eax
        mov     rdx, QWORD PTR [rdx]
        mov     rbx, rcx
        mov     QWORD PTR [rcx], rax
        mov     BYTE PTR [rcx], al
        mov     QWORD PTR [rcx+16], rax
        mov     QWORD PTR [rcx+24], 15
        call    std::basic_string<char,std::char_traits<char>,std::allocator<char> > & std::basic_string<char,std::char_traits<char>,std::allocator<char> >::assign(char const * const,unsigned __int64) ; std::basic_string<char,std::char_traits<char>,std::allocator<char> >::assign
        mov     rax, rbx
        add     rsp, 32                             ; 00000020H
        pop     rbx
        ret     0
X::X(string_view,use_data_size_tag) ENDP    ; X::X


========================================================

.LC0:
        .string "basic_string::_M_construct null not valid"
X::X(string_view, use_to_string_tag) [base object constructor]:
        push    r12
        mov     rcx, rsi
        mov     r12, rdx
        mov     rax, rdx
        push    rbp
        mov     rbp, rsi
        push    rbx
        mov     rbx, rdi
        add     rdi, 16
        sub     rsp, 16
        add     rcx, rdx
        mov     QWORD PTR [rbx], rdi
        je      .L2
        test    rsi, rsi
        je      .L18
.L2:
        mov     QWORD PTR [rsp+8], r12
        cmp     r12, 15
        ja      .L19
        cmp     r12, 1
        jne     .L5
        movzx   edx, BYTE PTR [rbp+0]
        mov     BYTE PTR [rbx+16], dl
.L6:
        mov     QWORD PTR [rbx+8], rax
        mov     BYTE PTR [rdi+rax], 0
        add     rsp, 16
        pop     rbx
        pop     rbp
        pop     r12
        ret
.L5:
        test    r12, r12
        je      .L6
        jmp     .L4
.L19:
        mov     rdi, rbx
        xor     edx, edx
        lea     rsi, [rsp+8]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned long&, unsigned long)
        mov     QWORD PTR [rbx], rax
        mov     rdi, rax
        mov     rax, QWORD PTR [rsp+8]
        mov     QWORD PTR [rbx+16], rax
.L4:
        mov     rdx, r12
        mov     rsi, rbp
        call    memcpy
        mov     rax, QWORD PTR [rsp+8]
        mov     rdi, QWORD PTR [rbx]
        jmp     .L6
.L18:
        mov     edi, OFFSET FLAT:.LC0
        call    std::__throw_logic_error(char const*)
X::X(string_view, use_data_size_tag) [base object constructor]:
        push    r12
        mov     rcx, rsi
        mov     r12, rdx
        mov     rax, rdx
        push    rbp
        mov     rbp, rsi
        push    rbx
        mov     rbx, rdi
        add     rdi, 16
        sub     rsp, 16
        add     rcx, rdx
        mov     QWORD PTR [rbx], rdi
        je      .L21
        test    rsi, rsi
        je      .L36
.L21:
        mov     QWORD PTR [rsp+8], r12
        cmp     r12, 15
        ja      .L37
        cmp     r12, 1
        jne     .L24
        movzx   edx, BYTE PTR [rbp+0]
        mov     BYTE PTR [rbx+16], dl
.L25:
        mov     QWORD PTR [rbx+8], rax
        mov     BYTE PTR [rdi+rax], 0
        add     rsp, 16
        pop     rbx
        pop     rbp
        pop     r12
        ret
.L24:
        test    r12, r12
        je      .L25
        jmp     .L23
.L37:
        mov     rdi, rbx
        xor     edx, edx
        lea     rsi, [rsp+8]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned long&, unsigned long)
        mov     QWORD PTR [rbx], rax
        mov     rdi, rax
        mov     rax, QWORD PTR [rsp+8]
        mov     QWORD PTR [rbx+16], rax
.L23:
        mov     rdx, r12
        mov     rsi, rbp
        call    memcpy
        mov     rax, QWORD PTR [rsp+8]
        mov     rdi, QWORD PTR [rbx]
        jmp     .L25
.L36:
        mov     edi, OFFSET FLAT:.LC0
        call    std::__throw_logic_error(char const*)
```